numpy/_core/src/umath/scalarmath.c.src

```diff
@@ -1393,6 +1381,109 @@ static PyObject *
 
 /**end repeat**/
 
+
+/**begin repeat
+ *
+ * #name = byte, ubyte, short, ushort, int, uint,
+ *             long, ulong, longlong, ulonglong#
+ * #Name = Byte, UByte, Short, UShort, Int, UInt,
+ *             Long, ULong,LongLong,ULongLong#
+ * #NAME = BYTE, UBYTE, SHORT, USHORT, INT, UINT,
+ *              LONG, ULONG, LONGLONG, ULONGLONG#
+ * #type = npy_byte, npy_ubyte, npy_short, npy_ushort, npy_int, npy_uint,
+ *             npy_long, npy_ulong, npy_longlong, npy_ulonglong#
+ */
+static PyObject *
+@name@_true_divide(PyObject *a, PyObject *b)
+{
+    /*
+     * See the more generic version above for additional comments, this
+     * differs in casting the arguments to double early on and in converting
+     * Python int and float directly to float.
+     */
+    PyObject *ret;
+    npy_float64 arg1, arg2, other_val;
+    @type@ other_val_conv;
+
+    int is_forward;
+    if (Py_TYPE(a) == &Py@Name@ArrType_Type) {
+        is_forward = 1;
+    }
+    else if (Py_TYPE(b) == &Py@Name@ArrType_Type) {
+        is_forward = 0;
+    }
+    else {
+        /* subclasses are involved */
+        is_forward = PyArray_IsScalar(a, @Name@);
+        assert(is_forward || PyArray_IsScalar(b, @Name@));
+    }
+
+    PyObject *other = is_forward ? b : a;
+
+    npy_bool may_need_deferring;
+    conversion_result res = convert_to_@name@(
+            other, &other_val_conv, &may_need_deferring);
+    other_val = other_val_conv;  /* Need a float value */
+    if (res == CONVERSION_ERROR) {
+        return NULL;  /* an error occurred (should never happen) */
+    }
+    if (may_need_deferring) {
+        BINOP_GIVE_UP_IF_NEEDED(a, b, nb_true_divide, @name@_true_divide);
+    }
+    switch (res) {
+        case DEFER_TO_OTHER_KNOWN_SCALAR:
+            Py_RETURN_NOTIMPLEMENTED;
+        case CONVERSION_SUCCESS:
+            break;  /* successfully extracted value we can proceed */
+        case OTHER_IS_UNKNOWN_OBJECT:
+        case PROMOTION_REQUIRED:
+            return PyGenericArrType_Type.tp_as_number->nb_true_divide(a,b);
+        case CONVERT_PYSCALAR:
+            /* This is the special behavior, convert to float64 directly */
+            if (DOUBLE_setitem(other, (char *)&other_val, NULL) < 0) {
+                return NULL;
+            }
+            break;
+        default:
+            assert(0);  /* error was checked already, impossible to reach */
+            return NULL;
+    }
+
+    npy_clear_floatstatus_barrier((char*)&arg1);
+
+    if (is_forward) {
+        arg1 = PyArrayScalar_VAL(a, @Name@);
+        arg2 = other_val;
+    }
+    else {
+        arg1 = other_val;
+        arg2 = PyArrayScalar_VAL(b, @Name@);
+    }
+
+    /*
+     * Prepare the actual calculation.
+     */
+    npy_float64 out;
+
+    int retstatus = @name@_ctype_true_divide(arg1, arg2, &out);
```

Comment:
Can this just be `ctype_true_divide` now? But are both in fact `double` now? I don't think so... Might be more logical to define just remove the whole definition of the function and do
```
*out = (npy_double)arg1 / (npy_double)arg2;
```
here

Reply:
Ah no, but you are right, I can delete the true divide integer version and just call  `double_dtype_true_divide`.

Reply:
Do you want to make a quick follow-up?

Reply:
I can, but only after my meeting is done ;).